trunk/source/Library-ModelCheckerUtils/src/de/uni_freiburg/informatik/ultimate/lib/modelcheckerutils/cfg/structure/IIcfg.java

```diff
@@ -71,6 +71,17 @@ public interface IIcfg<LOC extends IcfgLocation> extends IElement, IVisualizable
 	 */
 	Map<String, Set<LOC>> getProcedureErrorNodes();
 
+	/**
+	 * In our CFG we have distinct nodes that we call location of interest (LOI).
+	 * <li>LOIs are nodes for which verification algorithms typically produce invariants and for which counterexamples
+	 * have to provide values.
+	 * <li>ICFG transformations should try to preserve LOIs or provide a backtranslation that can compute invariants or
+	 * values of counterexample states for LOIs.
+	 *
+	 * @return all locations of interest (LOI).
+	 */
+	Set<LOC> getLocationsOfInterest();
```

Comment:
:+1:

Reply:
:+1:

Depending on the resolution of Frank's question above, please also document whether loop heads are (typically / always) assumed to be a subset of / distinct from LOIs, or neither.

Reply:
https://github.com/ultimate-pa/ultimate/pull/713/commits/d77a2221069b0ab5ac4f1a362c255c2f0b7837ca